app/models/weather_request.rb

```diff
@@ -3,8 +3,47 @@ class WeatherRequest < ApplicationRecord
 
   STATUSES = [PROCESSING = 'Processing',
               ERROR      = 'Error',
-              SUCCESS    = 'Success']
+              SUCCESS    = 'Success'].freeze
 
   validates :status, inclusion: { in: STATUSES }
-  
+
+  def weather_info
+    return nil if status != SUCCESS
+    hash_data = returned_json_as_hash
+    wind      = hash_data.dig('query', 'results', 'channel', 'wind')
+    astronomy = hash_data.dig('query', 'results', 'channel', 'astronomy')
+    condition = hash_data.dig('query', 'results', 'channel', 'item', 'condition')
+    OpenStruct.new(weather_time: condition['date'],
+                   temperature:  fahrenheit_to_celsius(condition['temp'].to_f),
+                   feels_like:   fahrenheit_to_celsius(wind['chill'].to_f),
```

Comment:
What happens if `'temp'` is missing in `condition` or `'chill'` is missing in `wind`?

Reply:
In my testing of the api, those values were always returned if the location was found, so in WeatherRequest, if the status is SUCCESS, I assume all the data is there.  The checking of the data is done in WeatherRequestService, and if it doesn't look ok, it uses an ERROR status.

Reply:
That's cool.  I like to isolate things to a reasonably granular level.  Basically, I don't like to make assumptions; mainly because someone might change the definition of how things work.  Thinking at a more granular level help insulate from such changes.

In this case, if `wind['chill']` is `nil`, then `feels_like` will be `-18`.  Is that OK?  Is it better to try to represent that that datum is missing, or just leave it?  Again, not a big deal.

Reply:
For a coding test, I thought I was being a little excessive with the error checking, lol.  I know I should be giving an example of production ready work, but I also had limited time.  If this was for production, I would probably do a lot more error checking in the WeatherRequestService, and then just output the "good" values to a model/table, so there would be no more checking to do.

Reply:
Of course, I don't think you need to implement fully granular checking for a coding test.  Value can come from the discussion.  I understand (and know) that you usually don't have a lot of time for coding tests, when you've got other work to do.  That's why the discussion is important (it saves time ;-) ).

Reply:
👍